src/org/hammerhead226/sharkmacro/motionprofiles/ProfileRecorder.java

```diff
@@ -68,9 +73,14 @@
 	 * @param right
 	 *            the right Talon
 	 */
-	public ProfileRecorder(TalonSRX left, TalonSRX right) {
+	public ProfileRecorder(TalonSRX left, TalonSRX right, RecordingType type) {
 		talons = new TalonSRX[] { left, right };
 		thread = new Notifier(new PeriodicRunnable());
+		if (type == RecordingType.VOLTAGE) {
+			this.recordingVoltage = true;
+		} else {
+			this.recordingVoltage = false;
+		}
```

Comment:
Could you just have a class property of `RecordingType recordingType`? Seems more extendable.